Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -278,13 +278,16 @@ struct ParametersAndReturnValidator {
     
     /// Checks if the symbol's documentation is inherited from another source location.
     private func hasInheritedDocumentationComment(symbol: UnifiedSymbolGraph.Symbol) -> Bool {
-        let symbolLocationURI = symbol.documentedSymbol?.mixins.getValueIfPresent(for: SymbolGraph.Symbol.Location.self)?.uri
-        for case .sourceCode(let location, _) in docChunkSources {
-            // Check if the symbol has documentation from another source location
-            return location?.uri != symbolLocationURI
+        guard let documentedSymbol = symbol.documentedSymbol else {
+            // If there's no doc comment, any documentation comes from an extension file and isn't inherited from another symbol.
+            return false
         }
-        // If the symbol didn't have any in-source documentation, check if there's a extension file override.
-        return docChunkSources.isEmpty
+        
+        // A symbol has inherited documentation if the doc comment doesn't come from the current module.
+        let moduleNames: Set<String> = symbol.modules.values.reduce(into: []) { $0.insert($1.name) }
```

Comment:
Why do we need to use a set here? We could write this line in a much simpler way using an array:

```suggestion
        let moduleNames = symbol.modules.values.map(\.name)
```

Also, I'm not sure how or why Swift casted to a set in your original version. Doesn't `reduce(into:_:)` return the same type as the accumulator? So in this case that would be an array: `into: []`. Maybe there's a Set initializer that takes an array that Swift is implicitly calling?
```

Reply:
I used a set here because I wanted to avoid checking the same module names more than once. 

If DocC is passed symbol graph files for different platforms version of a single module, they all have the same module name which would make this end up something like `["ModuleName", "ModuleName", "ModuleName", ...]` if it was an array.

You're right that `reduce(into:_:)` returns the same type as the accumulator, but `[]` can be any type that conform to to [`ExpressibleByArrayLiteral`](https://developer.apple.com/documentation/swift/expressiblebyarrayliteral) (which `Set` does). The explicit type annotation for the variable is one way to tell the compiler how it should interpret `[]` in that expression. 

I could have also done 
```swift
let moduleNames = Set(symbol.modules.values.map(\.name))
```
but it would have the small downside of first creating an array with duplicate values and then creating a set from those values.

Reply:
I could have also done 
```swift
let moduleNames = symbol.modules.values.reduce(into: Set()) { $0.insert($1.name) }
```
which perhaps is less confusing syntax

Reply:
Thanks for the explanation! - I wasn't familiar with `ExpressibleByArrayLiteral` and how Swift uses that in a practical manner like this.